CRM/Upgrade/Incremental/Base.php

```diff
@@ -156,12 +156,12 @@ public static function addColumn($ctx, $table, $column, $properties, $localizabl
     $domain = new CRM_Core_DAO_Domain();
     $domain->find(TRUE);
     $queries = [];
-    if (!CRM_Core_BAO_SchemaHandler::checkIfFieldExists($table, $column)) {
+    if (!CRM_Core_BAO_SchemaHandler::checkIfFieldExists($table, $column, FALSE)) {
```

Comment:
Use-cases:

| # | Lang config? | Column already exists? |Column is localizable?|
|---|---|---|---|
| 1 | Single Lang |AlreadyExists|Localizable
| 2 | Single Lang |AlreadyExists|Normal
| 3 | Single Lang|NewField|Localizable
| 4 | Single Lang|NewField|Normal
| 5 | Multi Lang|AlreadyExists|Localizable
| 6 | Multi Lang|AlreadyExists|Normal
| 7| Multi Lang|NewField|Localizable
| 8| Multi Lang|NewField|Normal

It reads as if L159 is supposed to determine if the column already exists. I'm not sure this check is right. Suppose:

* We were running addColumn() with a localized column like`civicrm_event.confirm_email_text`.
* The current config is MultiLang (en_US+fr_FR) and the column AlreadyExists

The actual columns in the DB would be named `confirm_email_text_en_US` and `confirm_email_text_fr_FR`. L159 would test for `civicrm_event.confirm_email_text`, but that would never exist in the table when using multilingual schema? So if you upgrade a multilingual site that already has the column, won't it try to re-add the column and produce a failure?

Reply:
@totten the column name is the non localised version of the column name, if we are on a localised database this will then be shown by the fact that it will be halted at the check L164

Reply:
so we test for the non localized version first, if its not there, then if it is a database that is to be localized -> check for the localized version if the field is to be localized

Reply:
Ah, yeah. :+1: